### PR TITLE
issue 523.1: address build issues, ensure FlushFeature subscriptions are cleared during stop(), assert that they are deallocated

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/RestServer/FlushFeature.cpp [hunk 1]
```diff
@@ -765,6 +799,14 @@ void FlushFeature::stop() {
       _isRunning.store(false);
       _flushThread.reset();
     }
+
+    {
+      std::lock_guard<std::mutex> lock(_flushSubscriptionsMutex);
+
+      // release any remamining flush subscritions so that they may get deallocated ASAP
```
Comment (graetzer) [inline arangod/RestServer/FlushFeature.cpp]:
```suggestion
      // release any remamining flush subscriptions so that they may get deallocated ASAP
```

Comment (vasiliy-arangodb) [reply arangod/RestServer/FlushFeature.cpp]: done, thank you for catching that

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3117/

**Comment by KVS85** [comment]
> http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3117/

It's still not compiling in Enterprise.

**Comment by vasiliy-arangodb** [comment]
@KVS85 
> It's still not compiling in Enterprise.
where?
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3117/
is now even green (not blue anymore)

**Comment by KVS85** [comment]
@vasiliy-arangodb you should run http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix/ also to validate this PR.

It didn't compile there when I launched.

**Comment by vasiliy-arangodb** [comment]
```
@KVS85 
> you should run http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix/ also to validate this PR.
>
> It didn't compile there when I launched.
http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix/224/ just finished, failing only on tests/js/server/resilience/moving-shards-cluster.js
"testMoveShardFromLeaderRepl_data"
I'll rerun again since IIRC that i one of the random failing tests
```

**Comment by KVS85** [comment]
No need to rerun then if compilation is OK now.

**Comment by vasiliy-arangodb** [comment]
@KVS85 full green success http://jenkins01.arangodb.biz:8080/view/3.devel/job/arangodb-devel-mac-matrix/225/

**Pull request merged**